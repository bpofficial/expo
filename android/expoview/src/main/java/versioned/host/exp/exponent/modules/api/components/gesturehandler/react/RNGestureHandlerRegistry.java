package versioned.host.exp.exponent.modules.api.components.gesturehandler.react;

import android.util.SparseArray;
import android.view.View;

import versioned.host.exp.exponent.modules.api.components.gesturehandler.GestureHandler;
import versioned.host.exp.exponent.modules.api.components.gesturehandler.GestureHandlerRegistry;

import java.util.ArrayList;

public class RNGestureHandlerRegistry implements GestureHandlerRegistry {

  private SparseArray<ArrayList<GestureHandler>> mHandlers = new SparseArray<>();

  public void registerHandlerForViewWithTag(int viewTag, GestureHandler handler) {
    ArrayList<GestureHandler> listToAdd = mHandlers.get(viewTag);
    if (listToAdd == null) {
      listToAdd = new ArrayList<>(1);
      listToAdd.add(handler);
      mHandlers.put(viewTag, listToAdd);
    } else {
      listToAdd.add(handler);
    }
  }

  public void dropHandlersForViewWithTag(int viewTag) {
    mHandlers.delete(viewTag);
  }

  public void dropAllHandlers() {
    mHandlers.clear();
  }

  public ArrayList<GestureHandler> getHandlersForViewWithTag(int viewTag) {
    return mHandlers.get(viewTag);
  }

  @Override
  public ArrayList<GestureHandler> getHandlersForView(View view) {
    return getHandlersForViewWithTag(view.getId());
  }
}
