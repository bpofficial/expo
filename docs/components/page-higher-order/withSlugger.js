import * as React from 'react';

export const SluggerContext = React.createContext();

const withSlugger = (Component) => {
  const SluggerHOC = props => (
    <SluggerContext.Consumer>
      {sluggerInstance => <Component slugger={sluggerInstance} {...props} />}
    </SluggerContext.Consumer>
  );

  return SluggerHOC;
};

export default withSlugger;
