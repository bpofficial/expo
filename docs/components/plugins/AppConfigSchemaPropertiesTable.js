import * as React from 'react';
import { css } from 'react-emotion';
import { InlineCode } from '~/components/base/code';
import MDX from '@mdx-js/runtime';
import * as components from '~/common/translate-markdown';

import { expoColors } from '~/common/constants';

const STYLES_TABLE = css`
  font-size: 1rem;
  margin-top: 24px;
`;

const STYLES_HEAD = css`
  background-color: ${expoColors.gray[100]};
`;

const STYLES_DESCRIPTION_CELL = css`
  word-break: break-word;
  white-space: break-spaces;
`;

export function formatSchema(rawSchema) {
  let formattedSchema = [];

  //appends each schema property (each index will become a tablerow)
  rawSchema.map(property => {
    appendProperty(formattedSchema, property, 0);
  });

  return formattedSchema;
}

//appends a property and recursivley calls itself to append sub-properties, accounting for nested "level"
function appendProperty(formattedSchema, property, _level) {
  let level = _level;
  const propertyKey = property[0];
  const propertyValue = property[1];

  //don't append deprecated or "hidden" properties
  if (propertyValue.meta && (propertyValue.meta.deprecated || propertyValue.meta.hidden)) {
    return;
  }

  //append passed-in property
  formattedSchema.push({
    //The ` backticks are for markdown highlighting, and #### is to apply anchor linking only on top-level properties
    name: level ? `\`${propertyKey}\`` : `#### \`${propertyKey}\``,
    type: propertyValue.enum ? 'enum' : propertyValue.type,
    description: createDescription(property),
    level: level,
  });

  //increase nesting level for sub-properties
  level++;

  //recursively apply appending logic for each sub-property (if any) -> Note: sub-props are sometimes nested within "items"
  if (propertyValue.properties) {
    Object.entries(propertyValue.properties).forEach(subproperty => {
      appendProperty(formattedSchema, subproperty, level);
    });
  } else if (propertyValue.items && propertyValue.items.properties) {
    Object.entries(propertyValue.items.properties).forEach(subproperty => {
      appendProperty(formattedSchema, subproperty, level);
    });
  }
}

//setting up a property's formatted description value, with all the possible extra values
export function createDescription(property) {
  const propertyValue = property[1];

  let propertyDescription = propertyValue.description;
  if (propertyValue.exampleString) {
    propertyDescription += `\n\n>` + propertyValue.exampleString;
  }
  if (propertyValue.meta && propertyValue.meta.regexHuman) {
    propertyDescription += `\n\n` + propertyValue.meta.regexHuman;
  }
  if (propertyValue.meta && propertyValue.meta.expoKit) {
    propertyDescription += `\n>**ExpoKit**: ` + propertyValue.meta.expoKit;
  }
  if (propertyValue.meta && propertyValue.meta.bareWorkflow) {
    if (propertyValue.meta.expoKit || propertyValue.exampleString) {
      propertyDescription += `\n`;
    }
    propertyDescription += `\n>**Bare workflow**: ` + propertyValue.meta.bareWorkflow;
  }

  return propertyDescription;
}

export default class AppConfigSchemaPropertiesTable extends React.Component {
  render() {
    var rawSchema = Object.entries(this.props.schema);
    var formattedSchema = formatSchema(rawSchema);

    return (
      <table className={STYLES_TABLE}>
        <thead className={STYLES_HEAD}>
          <tr>
            <td>Property</td>
            <td>Type</td>
            <td>Description</td>
          </tr>
        </thead>
        <tbody>
          {formattedSchema.map((property, index) => {
            return (
              <tr key={index}>
                <td>
                  <div
                    data-testid={property.name}
                    style={{
                      marginLeft: `${12 + property.level * 32}px`,
                      display: property.level ? 'list-item' : 'block',
                      listStyleType: property.level % 2 ? 'default' : 'circle',
                      width: 'fit-content',
                      overflowX: 'visible',
                    }}>
                    <MDX components={components}>{property.name}</MDX>
                  </div>
                </td>
                <td>
                  <InlineCode>{property.type && property.type.toString()}</InlineCode>
                </td>
                <td className={STYLES_DESCRIPTION_CELL}>
                  <MDX components={components}>{property.description}</MDX>
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    );
  }
}
