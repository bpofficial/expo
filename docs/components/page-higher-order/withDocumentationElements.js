import * as React from 'react';
import GithubSlugger from 'github-slugger';
import { withRouter } from 'next/router';
import { MDXProvider } from '@mdx-js/tag';

import DocumentationPage from '~/components/DocumentationPage';
import { SluggerContext } from '~/components/page-higher-order/withSlugger';
import * as components from '~/common/translate-markdown';

const withDocumentationElements = (meta) => {
  const DocumentationElementsHOC = withRouter((props) => {
    const { router } = props;

    return (
      <DocumentationPage
        title={meta.title}
        url={router}
        asPath={router.asPath}
        sourceCodeUrl={meta.sourceCodeUrl}>
        <SluggerContext.Provider value={new GithubSlugger()}>
          <MDXProvider components={components}>{props.children}</MDXProvider>
        </SluggerContext.Provider>
      </DocumentationPage>
    )
  })

  return DocumentationElementsHOC;
};

export default withDocumentationElements;
