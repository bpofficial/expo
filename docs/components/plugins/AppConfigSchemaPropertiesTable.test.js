import { render } from '@testing-library/react';
import AppConfigSchemaPropertiesTable, {
  formatSchema,
  createDescription,
} from './AppConfigSchemaPropertiesTable';
import { SluggerContext } from '~/components/page-higher-order/withSlugger';
import GithubSlugger from 'github-slugger';

var testSchema = {
  name: {
    description: 'Name of your app.',
    type: 'string',
    meta: {
      bareWorkflow: "Edit the 'Display Name' field in Xcode",
    },
  },
  androidNavigationBar: {
    description: 'Configuration for the bottom navigation bar on Android.',
    type: 'object',
    properties: {
      visible: {
        description: 'Determines how and when the navigation bar is shown.',
        type: 'string',
        properties: {
          always: {
            description: 'Test sub-sub-property',
            type: 'boolean',
          },
        },
        meta: {
          expoKit: 'Set this property using Xcode.',
        },
        enum: ['leanback', 'immersive', 'sticky-immersive'],
      },
      backgroundColor: {
        description: 'Specifies the background color of the navigation bar. ',
        type: 'string',
        pattern: '^#|(&#x23;)\\d{6}$',
        meta: {
          regexHuman: "6 character long hex color string, eg: `'#000000'`",
        },
      },
    },
    meta: {
      expoKit: 'Set this property using AppConstants.java.',
      bareWorkflow: 'Set this property using just Xcode',
    },
  },
  intentFilters: {
    description: 'Configuration for setting an array of custom intent filters in Android manifest.',
    example: [
      {
        autoVerify: true,
        data: {
          host: '*.expo.io',
        },
      },
    ],
    exampleString: '\n [{ \n "autoVerify": true, \n "data": {"host": "*.expo.io" \n } \n }]',
    type: 'array',
    uniqueItems: true,
    items: {
      type: 'object',
      properties: {
        autoVerify: {
          description:
            'You may also use an intent filter to set your app as the default handler for links',
          type: 'boolean',
        },
        data: {
          type: ['array', 'object'],
          items: {
            type: 'object',
            properties: {
              host: { description: 'the host, e.g. `myapp.io`', type: 'string' },
            },
            additionalProperties: false,
          },
          properties: {
            host: { type: 'string' },
          },
          additionalProperties: false,
        },
      },
      additionalProperties: false,
      required: ['action'],
    },
    meta: {
      bareWorkflow: 'This is set in AndroidManifest.xml directly.',
    },
  },
};

describe('AppConfigSchemaPropertiesTable', () => {
  test('correctly matches snapshot', () => {
    const { container } = render(
      <SluggerContext.Provider value={new GithubSlugger()}>
        <AppConfigSchemaPropertiesTable schema={testSchema} />
      </SluggerContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});

describe('formatSchema', () => {
  const formattedSchema = formatSchema(Object.entries(testSchema));
  test('name is property level 0', () => {
    expect(formattedSchema[0].level).toBe(0);
  });
  test('androidNavigationBar is property level 0', () => {
    expect(formattedSchema[1].level).toBe(0);
  });
  test('visible is subproperty level 1', () => {
    expect(formattedSchema[2].level).toBe(1);
  });
  test('visible has type "enum"', () => {
    expect(formattedSchema[2].type).toBe('enum');
  });
  test('always is subproperty level 2', () => {
    expect(formattedSchema[3].level).toBe(2);
  });
  test('backgroundColor is subproperty level 1', () => {
    expect(formattedSchema[4].level).toBe(1);
  });
  test('intentFilters is property level 0', () => {
    expect(formattedSchema[5].level).toBe(0);
  });
  test('autoVerify is subproperty level 1', () => {
    expect(formattedSchema[6].level).toBe(1);
  });
  test('data is subproperty level 1', () => {
    expect(formattedSchema[7].level).toBe(1);
  });
  test('host is subproperty level 2', () => {
    expect(formattedSchema[8].level).toBe(2);
  });
});

describe('createDescription', () => {
  test('exampleString, bareWorkflow are both added correctly to intentFilters', () => {
    const intentFiltersObject = Object.entries(testSchema)[2];
    const intentFiltersObjectValue = intentFiltersObject[1];
    const result = createDescription(intentFiltersObject);

    expect(result).toBe(
      `${intentFiltersObjectValue.description}\n\n>${intentFiltersObjectValue.exampleString}\n\n>**Bare workflow**: ${intentFiltersObjectValue.meta.bareWorkflow}`
    );
  });

  test('regexHuman is added correctly to backgroundColor', () => {
    //Note: to access this subproperty is tedious without a call to formatSchema
    const backgroundColorObject = Object.entries(Object.values(testSchema)[1].properties)[1];
    const backgroundColorObjectValue = backgroundColorObject[1];
    const result = createDescription(backgroundColorObject);

    expect(result).toBe(
      `${backgroundColorObjectValue.description}\n\n${backgroundColorObjectValue.meta.regexHuman}`
    );
  });

  test('expoKit is added correctly to visible', () => {
    //Note: to access this subproperty is tedious without a call to formatSchema
    const visibleObject = Object.entries(Object.values(testSchema)[1].properties)[0];
    const visibleObjectValue = visibleObject[1];
    const result = createDescription(visibleObject);

    expect(result).toBe(
      `${visibleObjectValue.description}\n>**ExpoKit**: ${visibleObjectValue.meta.expoKit}`
    );
  });

  test('bareWorkflow is added correctly to name', () => {
    const nameObject = Object.entries(testSchema)[0];
    const nameObjectValue = nameObject[1];
    const result = createDescription(nameObject);

    expect(result).toBe(
      `${nameObjectValue.description}\n>**Bare workflow**: ${nameObjectValue.meta.bareWorkflow}`
    );
  });

  test('expoKit, bareWorkflow both added correctly to androidNavigationBar', () => {
    const androidNavigationBarObject = Object.entries(testSchema)[1];
    const androidNavigationBarObjectValue = androidNavigationBarObject[1];
    const result = createDescription(androidNavigationBarObject);

    expect(result).toBe(
      `${androidNavigationBarObjectValue.description}\n>**ExpoKit**: ${androidNavigationBarObjectValue.meta.expoKit}\n\n>**Bare workflow**: ${androidNavigationBarObjectValue.meta.bareWorkflow}`
    );
  });
});
