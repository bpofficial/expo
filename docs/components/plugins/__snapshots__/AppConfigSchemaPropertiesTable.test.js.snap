// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppConfigSchemaPropertiesTable correctly matches snapshot 1`] = `
<div>
  <table
    class="css-1ujlu4q"
  >
    <thead
      class="css-1gbkmmb"
    >
      <tr>
        <td>
          Property
        </td>
        <td>
          Type
        </td>
        <td>
          Description
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            data-testid="#### \`name\`"
            style="margin-left: 12px; display: block; list-style-type: circle; overflow-x: visible;"
          >
            <h4
              class="css-1l3zmkw"
              data-heading="true"
            >
              <div
                class="css-i07u6a"
              >
                <span
                  class="css-1fj428k"
                  id="name"
                />
                <a
                  class="permalink css-1lc60ja"
                  href="#name"
                >
                  <svg
                    aria-label="permalink"
                    class="anchor-icon"
                    version="1.1"
                    viewBox="0 0 11 11"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke="none"
                      stroke-width="1"
                    >
                      <g
                        fill="#9B9B9B"
                        transform="translate(-432.000000, -181.000000)"
                      >
                        <path
                          d="M442.36949,184.673746 L440.543368,186.500079 C440.047223,186.996013 439.369736,187.191548 438.72445,187.101937 L439.934732,185.891655 L441.760854,184.065321 C442.26505,183.561549 442.26505,182.743395 441.760854,182.238988 C441.256658,181.735003 440.438928,181.735003 439.934732,182.238988 L438.10861,184.065321 L436.897904,185.275815 C436.808293,184.630529 437.003616,183.95283 437.499762,183.456684 L439.325883,181.630775 C440.166281,180.789742 441.528881,180.789742 442.369278,181.630775 C443.209887,182.470748 443.209887,183.833772 442.36949,184.673746 L442.36949,184.673746 Z M438.717247,185.891655 L436.891125,187.717565 L436.282277,187.108928 L438.10861,185.282594 L438.717247,185.891655 L438.717247,185.891655 Z M433.238882,188.934838 C432.734686,189.439246 432.734686,190.256552 433.238882,190.760959 C433.743078,191.265579 434.560807,191.265579 435.065003,190.760959 L436.891125,188.934838 L438.101619,187.724344 C438.191442,188.369842 437.996119,189.047329 437.499974,189.543474 L435.673852,191.369384 C434.833455,192.210205 433.470854,192.210205 432.630457,191.369384 C431.789848,190.529411 431.789848,189.166387 432.630457,188.326413 L434.456579,186.500079 C434.952724,186.003722 435.630423,185.808399 436.275709,185.898222 L435.065215,187.108928 L433.238882,188.934838 L433.238882,188.934838 Z"
                        />
                      </g>
                    </g>
                  </svg>
                </a>
                <div
                  class="permalink-child"
                >
                  <code
                    class="css-7zpg9q inline"
                  >
                    name
                  </code>
                </div>
              </div>
            </h4>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            string
          </code>
        </td>
        <td
          class="css-12joc8l"
        >
          <div
            class="css-100y9e2 "
            data-text="true"
          >
            Name of your app.
          </div>
          <blockquote
            class="css-80nrkz"
            data-text="true"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <strong
                class="css-utowyr"
              >
                Bare workflow
              </strong>
              : Edit the 'Display Name' field in Xcode
            </div>
          </blockquote>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="#### \`androidNavigationBar\`"
            style="margin-left: 12px; display: block; list-style-type: circle; overflow-x: visible;"
          >
            <h4
              class="css-1l3zmkw"
              data-heading="true"
            >
              <div
                class="css-i07u6a"
              >
                <span
                  class="css-1fj428k"
                  id="androidnavigationbar"
                />
                <a
                  class="permalink css-1lc60ja"
                  href="#androidnavigationbar"
                >
                  <svg
                    aria-label="permalink"
                    class="anchor-icon"
                    version="1.1"
                    viewBox="0 0 11 11"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke="none"
                      stroke-width="1"
                    >
                      <g
                        fill="#9B9B9B"
                        transform="translate(-432.000000, -181.000000)"
                      >
                        <path
                          d="M442.36949,184.673746 L440.543368,186.500079 C440.047223,186.996013 439.369736,187.191548 438.72445,187.101937 L439.934732,185.891655 L441.760854,184.065321 C442.26505,183.561549 442.26505,182.743395 441.760854,182.238988 C441.256658,181.735003 440.438928,181.735003 439.934732,182.238988 L438.10861,184.065321 L436.897904,185.275815 C436.808293,184.630529 437.003616,183.95283 437.499762,183.456684 L439.325883,181.630775 C440.166281,180.789742 441.528881,180.789742 442.369278,181.630775 C443.209887,182.470748 443.209887,183.833772 442.36949,184.673746 L442.36949,184.673746 Z M438.717247,185.891655 L436.891125,187.717565 L436.282277,187.108928 L438.10861,185.282594 L438.717247,185.891655 L438.717247,185.891655 Z M433.238882,188.934838 C432.734686,189.439246 432.734686,190.256552 433.238882,190.760959 C433.743078,191.265579 434.560807,191.265579 435.065003,190.760959 L436.891125,188.934838 L438.101619,187.724344 C438.191442,188.369842 437.996119,189.047329 437.499974,189.543474 L435.673852,191.369384 C434.833455,192.210205 433.470854,192.210205 432.630457,191.369384 C431.789848,190.529411 431.789848,189.166387 432.630457,188.326413 L434.456579,186.500079 C434.952724,186.003722 435.630423,185.808399 436.275709,185.898222 L435.065215,187.108928 L433.238882,188.934838 L433.238882,188.934838 Z"
                        />
                      </g>
                    </g>
                  </svg>
                </a>
                <div
                  class="permalink-child"
                >
                  <code
                    class="css-7zpg9q inline"
                  >
                    androidNavigationBar
                  </code>
                </div>
              </div>
            </h4>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            object
          </code>
        </td>
        <td
          class="css-12joc8l"
        >
          <div
            class="css-100y9e2 "
            data-text="true"
          >
            Configuration for the bottom navigation bar on Android.
          </div>
          <blockquote
            class="css-80nrkz"
            data-text="true"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <strong
                class="css-utowyr"
              >
                ExpoKit
              </strong>
              : Set this property using AppConstants.java.
            </div>
          </blockquote>
          <blockquote
            class="css-80nrkz"
            data-text="true"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <strong
                class="css-utowyr"
              >
                Bare workflow
              </strong>
              : Set this property using just Xcode
            </div>
          </blockquote>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="\`visible\`"
            style="margin-left: 44px; display: list-item; list-style-type: default; overflow-x: visible;"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <code
                class="css-7zpg9q inline"
              >
                visible
              </code>
            </div>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            enum
          </code>
        </td>
        <td
          class="css-12joc8l"
        >
          <div
            class="css-100y9e2 "
            data-text="true"
          >
            Determines how and when the navigation bar is shown.
          </div>
          <blockquote
            class="css-80nrkz"
            data-text="true"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <strong
                class="css-utowyr"
              >
                ExpoKit
              </strong>
              : Set this property using Xcode.
            </div>
          </blockquote>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="\`always\`"
            style="margin-left: 76px; display: list-item; list-style-type: circle; overflow-x: visible;"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <code
                class="css-7zpg9q inline"
              >
                always
              </code>
            </div>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            boolean
          </code>
        </td>
        <td
          class="css-12joc8l"
        >
          <div
            class="css-100y9e2 "
            data-text="true"
          >
            Test sub-sub-property
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="\`backgroundColor\`"
            style="margin-left: 44px; display: list-item; list-style-type: default; overflow-x: visible;"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <code
                class="css-7zpg9q inline"
              >
                backgroundColor
              </code>
            </div>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            string
          </code>
        </td>
        <td
          class="css-12joc8l"
        >
          <div
            class="css-100y9e2 "
            data-text="true"
          >
            Specifies the background color of the navigation bar. 
          </div>
          <div
            class="css-100y9e2 "
            data-text="true"
          >
            6 character long hex color string, eg: 
            <code
              class="css-7zpg9q inline"
            >
              '#000000'
            </code>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="#### \`intentFilters\`"
            style="margin-left: 12px; display: block; list-style-type: circle; overflow-x: visible;"
          >
            <h4
              class="css-1l3zmkw"
              data-heading="true"
            >
              <div
                class="css-i07u6a"
              >
                <span
                  class="css-1fj428k"
                  id="intentfilters"
                />
                <a
                  class="permalink css-1lc60ja"
                  href="#intentfilters"
                >
                  <svg
                    aria-label="permalink"
                    class="anchor-icon"
                    version="1.1"
                    viewBox="0 0 11 11"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      stroke="none"
                      stroke-width="1"
                    >
                      <g
                        fill="#9B9B9B"
                        transform="translate(-432.000000, -181.000000)"
                      >
                        <path
                          d="M442.36949,184.673746 L440.543368,186.500079 C440.047223,186.996013 439.369736,187.191548 438.72445,187.101937 L439.934732,185.891655 L441.760854,184.065321 C442.26505,183.561549 442.26505,182.743395 441.760854,182.238988 C441.256658,181.735003 440.438928,181.735003 439.934732,182.238988 L438.10861,184.065321 L436.897904,185.275815 C436.808293,184.630529 437.003616,183.95283 437.499762,183.456684 L439.325883,181.630775 C440.166281,180.789742 441.528881,180.789742 442.369278,181.630775 C443.209887,182.470748 443.209887,183.833772 442.36949,184.673746 L442.36949,184.673746 Z M438.717247,185.891655 L436.891125,187.717565 L436.282277,187.108928 L438.10861,185.282594 L438.717247,185.891655 L438.717247,185.891655 Z M433.238882,188.934838 C432.734686,189.439246 432.734686,190.256552 433.238882,190.760959 C433.743078,191.265579 434.560807,191.265579 435.065003,190.760959 L436.891125,188.934838 L438.101619,187.724344 C438.191442,188.369842 437.996119,189.047329 437.499974,189.543474 L435.673852,191.369384 C434.833455,192.210205 433.470854,192.210205 432.630457,191.369384 C431.789848,190.529411 431.789848,189.166387 432.630457,188.326413 L434.456579,186.500079 C434.952724,186.003722 435.630423,185.808399 436.275709,185.898222 L435.065215,187.108928 L433.238882,188.934838 L433.238882,188.934838 Z"
                        />
                      </g>
                    </g>
                  </svg>
                </a>
                <div
                  class="permalink-child"
                >
                  <code
                    class="css-7zpg9q inline"
                  >
                    intentFilters
                  </code>
                </div>
              </div>
            </h4>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            array
          </code>
        </td>
        <td
          class="css-12joc8l"
        >
          <div
            class="css-100y9e2 "
            data-text="true"
          >
            Configuration for setting an array of custom intent filters in Android manifest.
          </div>
          <blockquote
            class="css-80nrkz"
            data-text="true"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
               
              [{
"autoVerify": true,
"data": {"host": "*.expo.io"
}
}]
            </div>
          </blockquote>
          <blockquote
            class="css-80nrkz"
            data-text="true"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <strong
                class="css-utowyr"
              >
                Bare workflow
              </strong>
              : This is set in AndroidManifest.xml directly.
            </div>
          </blockquote>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="\`autoVerify\`"
            style="margin-left: 44px; display: list-item; list-style-type: default; overflow-x: visible;"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <code
                class="css-7zpg9q inline"
              >
                autoVerify
              </code>
            </div>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            boolean
          </code>
        </td>
        <td
          class="css-12joc8l"
        >
          <div
            class="css-100y9e2 "
            data-text="true"
          >
            You may also use an intent filter to set your app as the default handler for links
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            data-testid="\`data\`"
            style="margin-left: 44px; display: list-item; list-style-type: default; overflow-x: visible;"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <code
                class="css-7zpg9q inline"
              >
                data
              </code>
            </div>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            array,object
          </code>
        </td>
        <td
          class="css-12joc8l"
        />
      </tr>
      <tr>
        <td>
          <div
            data-testid="\`host\`"
            style="margin-left: 76px; display: list-item; list-style-type: circle; overflow-x: visible;"
          >
            <div
              class="css-100y9e2 "
              data-text="true"
            >
              <code
                class="css-7zpg9q inline"
              >
                host
              </code>
            </div>
          </div>
        </td>
        <td>
          <code
            class="css-7zpg9q inline"
          >
            string
          </code>
        </td>
        <td
          class="css-12joc8l"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
